src/main/java/org/skife/jdbi/v2/MappingRegistry.java

```diff
@@ -49,32 +58,75 @@ public void add(ResultSetMapper mapper)
 
     public void add(ResultSetMapperFactory factory)
     {
-        factories.add(factory);
-        cache.clear();
+        rowFactories.add(factory);
+        rowCache.clear();
     }
 
     public ResultSetMapper mapperFor(Class type, StatementContext ctx) {
-        if (cache.containsKey(type)) {
-            ResultSetMapper mapper = cache.get(type);
+        if (rowCache.containsKey(type)) {
+            ResultSetMapper mapper = rowCache.get(type);
             if (mapper != null) {
                 return mapper;
             }
         }
 
-        for (ResultSetMapperFactory factory : factories) {
+        for (ResultSetMapperFactory factory : rowFactories) {
             if (factory.accepts(type, ctx)) {
                 ResultSetMapper mapper =  factory.mapperFor(type, ctx);
-                cache.put(type, mapper);
+                rowCache.put(type, mapper);
                 return mapper;
             }
         }
 
+        // TODO remove this and let the column mapper block below take over
         if (BUILT_IN_MAPPERS.accepts(type, ctx)) {
             ResultSetMapper mapper = BUILT_IN_MAPPERS.mapperFor(type, ctx);
-            cache.put(type, mapper);
+            rowCache.put(type, mapper);
+            return mapper;
+        }
+
+        ResultColumnMapper columnMapper = columnMapperFor(type, ctx);
+        if (columnMapper != null) {
+            ResultSetMapper mapper = new SingleColumnMapper(columnMapper);
+            rowCache.put(type, mapper);
             return mapper;
         }
 
         throw new DBIException("No mapper registered for " + type.getName()) {};
     }
+
+    public void addColumn(ResultColumnMapper mapper)
+    {
+        this.addColumn(new InferredColumnMapperFactory(mapper));
+    }
+
+    public void addColumn(ResultColumnMapperFactory factory) {
+        columnFactories.add(factory);
+        columnCache.clear();
+    }
+
+    public ResultColumnMapper columnMapperFor(Class type, StatementContext ctx) {
```

Comment:
That could work too.  I'm not a huge fan of suppressing warnings unless there truly is no way to get around them, but this really is a nit given that the existing code already is largely raw so I will defer to what you want to do.